internal/k8s/builder/gateway.go

```diff
@@ -370,33 +391,39 @@ type gwContainerCommandData struct {
 	SDSPort           int
 }
 
-// gwContainerCommandTpl is the template for the command executed by
-// the exec container.
-const gwContainerCommandTpl = `
-{{- if .ConsulCAFile}}
-export CONSUL_CACERT={{ .ConsulCAFile }}
-cat <<EOF >{{ .ConsulCAFile }}
-{{ .ConsulCAData }}
-EOF
-{{- end}}
-
-exec /bootstrap/consul-api-gateway exec -log-json \
-  -log-level {{ .LogLevel }} \
-  -gateway-host "{{ .GatewayHost }}" \
-  -gateway-name {{ .GatewayName }} \
+// gwContainerArgsTpl is the template for the command arguments executed in the Envoy container.
+// The resulting args are split on \n to obtain a []string for the pod spec's args.
+// Note: Make sure not to leave whitespace at the beginning or end of any line.
+const gwContainerArgsTpl = `
```

Comment:
Left the godoc here to explain since the template looks kinda weird now.

Basically, the flag name and its value are expected to be two separate items in the `args` array for the pod template. It was still easier to compile the list of args using a template like we were before since the other options is a big string builder with a bunch of conditionally-included `fmt.Sprintf`s.